Src/Common/HeadersUtilities.cs

```diff
@@ -0,0 +1,139 @@
+        public static string GetHeaderKeyValue(IEnumerable<string> headerValue, string keyName)
+        {
+            if (headerValue != null)
+            {
+                foreach (string keyNameValue in headerValue)
+                {
+                    string[] keyNameValueParts = keyNameValue.Trim().Split('=');
+                    if (keyNameValueParts.Length == 2 && keyNameValueParts[0].Trim() == keyName)
+                    {
+                        return keyNameValueParts[1].Trim();
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Given the provided list of header value strings, return a comma-separated list of key
+        /// name/value pairs with the provided keyName and keyValue. If the initial header value
+        /// strings contains the key name, then the original key value should be replaced with the
+        /// provided key value. If the initial header value strings don't contain the key name,
+        /// then the key name/value pair should be added to the comma-separated list and returned.
+        /// </summary>
+        /// <param name="headerValues">The existing header values that the key/value pair should be added to.</param>
+        /// <param name="keyName">The name of the key to add.</param>
+        /// <param name="keyValue">The value of the key to add.</param>
+        /// <returns>The result of setting the provided key name/value pair into the provided headerValues.</returns>
+        public static string SetHeaderKeyValue(IEnumerable<string> headerValues, string keyName, string keyValue)
+        {
+            StringBuilder result = new StringBuilder();
+
+            bool found = false;
+            if (headerValues != null)
+            {
+                foreach (string keyValuePair in headerValues)
+                {
+                    int equalsSignIndex = keyValuePair.IndexOf('=');
+                    if (equalsSignIndex == -1)
+                    {
+                        Append(result, keyValuePair.Trim());
+                    }
+                    else
+                    {
+                        string name = TrimSubstring(keyValuePair, 0, equalsSignIndex);
+                        if (name == null)
+                        {
+                            Append(result, keyValuePair.Trim());
+                        }
+                        else if (name != keyName)
+                        {
+                            string value = TrimSubstring(keyValuePair, equalsSignIndex + 1, keyValuePair.Length);
+                            if (value == null)
+                            {
+                                value = string.Empty;
+                            }
+                            Append(result, CreateKeyValuePair(name, value));
+                        }
+                        else if (!found)
```

Comment:
That changes the order, and I was trying to not modify your tests. If the order really doesn't matter, then yes, I can do that.